targets/TARGET_Analog_Devices/TARGET_ADUCM302X/TARGET_ADUCM3029/bsp/rng/adi_rng.c

```diff
@@ -0,0 +1,796 @@
+* Pm140 (rule 11.3): a cast should not be performed between a pointer type and an integral type
+*   The rule makes an exception for memory-mapped register accesses.
+*/
+#pragma diag_suppress=Pm011,Pm073,Pm143,Pm050
+#endif /* __ICCARM__ */
+
+#if   defined(__ADUCM4x50__) ||   defined(__ADUCM302x__)
+#define NUM_RNG_DEVICES       (1u)
+#else
+#error "Unsupported processor"
+#endif
+
+/*==============  D A T A  ===============*/
+
+/**
+ * Information for managing all the RNG devices available
+ */
+#ifdef __ICCARM__
+#pragma diag_suppress=Pm140
+#endif
+
+static ADI_RNG_DEV_TYPE gRNG_Device[NUM_RNG_DEVICES] =
+{
+  {(ADI_RNG_TypeDef*)pADI_RNG0,NULL} /* RNG0 */
+};
+#ifdef __ICCARM__
+#pragma diag_default=Pm140
+#endif
+
+/* Forward prototypes */
+void RNG_Int_Handler(void);
+
+/** Check the validity of a handle for debug mode */
+#ifdef ADI_DEBUG
+#define ADI_RNG_INVALID_HANDLE(h) (&gRNG_Device[0] != (h))
+#endif
+
+/*! \endcond */
+
+/*!
+    @brief Opena a Random Number Generator Device
+
+    @param[in]  nDeviceNum   Device number to be opened.
+    @param[in]  pMemory      Pointer to the memory to be used by the driver.
+                             Size of the memory should be at least #ADI_RNG_MEMORY_SIZE bytes.
+    @param[in]  MemorySize   Size of the memory passed in pMemory parameter.
+    @param[out] phDevice     Pointer to a location in the calling function memory space to which
+                             the device handle will be written upon successful driver initialization.
+
+    @return    Status
+               - #ADI_RNG_SUCCESS                      RNG device driver opened successfully.
+               - #ADI_RNG_INVALID_PARAM [D]            The memory passed to the API is either NULL or its size is not sufficient.
+               - #ADI_RNG_ALREADY_INITIALIZED [D]  The RNG is already initialized.
+               - #ADI_RNG_BAD_DEVICE_NUM [D]       The device number is invalid.
+
+    Initialize and allocate a RNG device for other use.  The core NVIC RNG interrupt is enabled.  This API
+    must preceed all other RNG API calls and the handle returned must be passed to all other RNG API calls.
+
+    @note   The contents of \a ppDevice will be set to NULL upon failure.\n\n
+
+    @note   The RNG device driver will clear all pending interrupts and disable all RNG
+            interrupts during RNG device initialization.
+
+    @sa    adi_rng_Close().
+*/
+ADI_RNG_RESULT adi_rng_Open(
+                            uint32_t             const  nDeviceNum,
+                            void*                const  pMemory,
+                            uint32_t             const  MemorySize,
+                            ADI_RNG_HANDLE*      const  phDevice
+                            )
+{
+    ADI_RNG_DEV_TYPE *pDevice;
+
+    /* store a bad handle in case of failure */
+    *phDevice = (ADI_RNG_HANDLE) NULL;
+
+#ifdef ADI_DEBUG
+    if (nDeviceNum >= NUM_RNG_DEVICES)
```

Comment:
Why are these check inside debug flag?

Reply:
I think these are for internal testing by the driver developers only.